Source/WebCore/platform/graphics/ImageFrameWorkQueue.cpp

```diff
@@ -86,16 +86,16 @@ void ImageFrameWorkQueue::start()
             }
 
             // Even if we fail to decode the frame, it is important to sync the main thread with this result.
-            callOnMainThread([protectedThis, protectedWorkQueue, protectedSource, request, nativeImage = WTFMove(nativeImage)] () mutable {
+            callOnMainThread([protectedThis, protectedWorkQueue, protectedSource, request, sourceURL = sourceURL.isolatedCopy(), nativeImage = WTFMove(nativeImage)] () mutable {
                 // The WorkQueue may have been recreated before the frame was decoded.
                 if (protectedWorkQueue.ptr() != protectedThis->m_workQueue || protectedSource.ptr() != protectedThis->m_source.get()) {
-                    LOG(Images, "ImageFrameWorkQueue::%s - %p - url: %s. WorkQueue was recreated at index = %d.", __FUNCTION__, protectedThis.ptr(), protectedSource->sourceUTF8(), request.index);
+                    LOG(Images, "ImageFrameWorkQueue::%s - %p - url: %s. WorkQueue was recreated at index = %d.", __FUNCTION__, protectedThis.ptr(), sourceURL.utf8().data(), request.index);
```

Comment:
Do we really need to copy sourceURL twice, just for logging? Can't we just get it from protectedSource again, or might it have changed?

Reply:
Yes it is true that we can get the sourceURL from protectedSource() as long as we make the call on the main thread. I was trying to log the decoding time on the decoding thread. So when adding the log message, a crash happened because of accessing the sourceURL String from another thread. Because currently there is no log message on the decoding thread, it is better to save the String copy time. So I removed this new copyIsolated() code.